sirepo/auth/ldap.py

```diff
@@ -60,7 +63,7 @@ def _validate_entry(creds, field):
             else:
                 return
             raise sirepo.util.UserAlert(
-                "invalid user and/or password",
+                "Invalid user and/or password",
```

Comment:
I think this should be an SReply as well. We want to use form_error vs UserAlert to show the user errors.

I would argue that it would be better style to do this:

```py
r = _validate_entry(res, "email") and _validate_entry(res, "password") and _bind(res)
if r:
     return r
```

And convert the raises into returns. There's no need to raise an exception in this case, and it's complicated. The api naturally returns an SReply, and you should implement it that way.

Reply:
Noting this style, makes much more sense for api